resources/js/Organisms/Form.tsx

```diff
@@ -32,106 +32,86 @@ export interface IFormState {
   submitSuccess?: boolean;
 }
 
-export class Form extends React.Component<IFormProps, IFormState> {
-  constructor(props: IFormProps) {
-    super(props);
-
-    const errors: IErrors = {};
-    const values: IValues = { fullname: "" };
-    this.state = {
-      errors,
-      values
-    };
-  }
-
-  /**
-   * Returns whether there are any errors in the errors object that is passed in
-   * @param {IErrors} errors - The field errors
-   */
-  private haveErrors(errors: IErrors) {
+const Form: React.FC<IFormProps> = ({ action, render }) => {
+  const [errors, setErrors] = useState<IErrors>({});
+  const [values, setValues] = useState<IValues>({ fullname: "" });
+  const [submitSuccess, setSuccess] = useState(undefined);
+
+  const haveErrors = (errors: IErrors) => {
     let haveError: boolean = false;
     Object.keys(errors).map((key: string) => {
       if (errors[key].length > 0) {
         haveError = true;
       }
     });
     return haveError;
-  }
+  };
 
-  private updateState = (target: HTMLInputElement) => {
+  const updateState = (target: HTMLInputElement) => {
     const { name, value } = target;
-    this.setState({ values: { [name]: value } });
+    setValues({ [name]: value });
+  };
+
+  const validateForm = (): boolean => {
+    // TODO - validate form
+    // (set errors? for now just set as empty object)
+    setErrors({});
+    return true;
   };
 
-  /**
-   * Handles form submission
-   * @param {React.FormEvent<HTMLFormElement>} e - The form event
-   */
-  private handleSubmit = async (
+  const submitForm = async (): Promise<boolean> => {
+    // TODO - submit the form
+    return true;
+  };
+
+  const handleSubmit = async (
     e: React.FormEvent<HTMLFormElement>
   ): Promise<void> => {
     e.preventDefault();
-    console.log(this.state.values);
+    console.log(values);
 
-    if (this.validateForm()) {
-      const submitSuccess: boolean = await this.submitForm();
-      this.setState({ submitSuccess });
+    if (validateForm()) {
```

Comment:
As `validateForm` is async, I think this would need to be awaited here to get the result, rather than the Promise.
Something like this:
```suggestion
    const formIsValid = await validateForm();
    if (formIsValid) {
```

As an aside - does `validateForm` need to be async? Are we expecting that it might call some APIs to check if it's valid?